File: client/src/app/modules/miscellaneous/services/quotation.abandon.reason.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { QuotationAbandonReason } from '../model/QuotationAbandonReason';

@Injectable({
  providedIn: 'root'
})
export class QuotationAbandonReasonService extends AppRestService<QuotationAbandonReason>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getQuotationAbandonReasons() {
    return this.getListCached(new HttpParams(), "quotation-abandon-reasons");
  }

  addOrUpdateQuotationAbandonReason(abandonReason: QuotationAbandonReason) {
    return this.addOrUpdate(new HttpParams(), "quotation-abandon-reason/quotation", abandonReason, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/quotation/service/QuotationAbandonReasonService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.QuotationAbandonReason;

public interface QuotationAbandonReasonService {

    public List<QuotationAbandonReason> getQuotationAbandonReasons();

    public QuotationAbandonReason getAbandonReason(Integer id);

    public QuotationAbandonReason addOrUpdateQutationAbandonReason(QuotationAbandonReason abandonReason);
}
